File: 1615-range-sum-of-sorted-subarray-sums/range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:

        subarray_sums = []
        modulo = (10 ** 9) + 7
        
        for i in range(len(nums)):
            for j in range(i, len(nums)):
                subarray = nums[i: j + 1]
                summ = sum(subarray)
                subarray_sums.append(summ)

        subarray_sums.sort()
        output = 0
        for i in range(left - 1, right):
            output += subarray_sums[i] 

        return output % modulo
        

File: 1615-range-sum-of-sorted-subarray-sums/range-sum-of-sorted-subarray-sums.js
var rangeSum = function(nums, n, left, right) {
    let subarraySums = []
    let modulo = (10 ** 9) + 7

    for (let i = 0; i < nums.length; i++) {
        for (let j = i; j < nums.length; j++) {
            let subarray = nums.slice(i, j + 1)
            let sum = subarray.reduce((a, b)=> a + b, 0)
            subarraySums.push(sum)
        }
    }    
    subarraySums.sort((a, b)=> a - b)
    
    let sum = 0
    for (let i = left - 1; i < right; i++) {
        sum += subarraySums[i]
    }

    return sum % modulo
};